source/web-service/flaskapp/routes/records.py

```diff
@@ -540,6 +549,72 @@ def entity_record(entity_id):
             return abort(response)
 
 
+# 'DELETE' method.
+@records.route("/<path:id>", methods=["DELETE"])
+def delete(id):
+
+    result_dict = {}
+
+    # Authentication
+    status = authenticate_bearer(request)
+    if status != status_ok:
+        response = construct_error_response(status)
+        return abort(response)
+
+    current_app.logger.debug("Authentication checked - DELETE request allowed.")
+
+    # Get record from DB
+    db_rec = get_record(id)
+
+    # No such record. Return {"id": None} (format identical to 'ingest/delete')
+    if db_rec is None:
+        result_dict[id] = "null"
+        return jsonify(result_dict), 200
+
+    # Process DELETE
+    with db.session.no_autoflush:
+        try:
+            process_activity(db_rec.id, Event.Delete)
+            record_delete(db_rec, None, True)
+
+        # Catch only OperationalError exception (e.g. DB is down)
+        except exc.OperationalError as e:
+            current_app.logger.error(e)
+            current_app.logger.critical(
+                "Critical failure writing the updated Record to DB"
+            )
+            db.session.rollback()
+            return abort(construct_error_response(status_db_save_error))
+
+        # Process RDF if applicable
+        if current_app.config["PROCESS_RDF"] is True:
+
+            # this is REST DELETE, so we don't have the full record, just the 'id'
+            rec = json.dumps({"id": id, "_delete": True})
+            graphstore_result = process_graphstore_record_set([rec])
```

Comment:
The full id is this plus the RDFidPrefix:

```
full_uri = f"{current_app.config['RDFidPrefix']/{id}"
current_app.logger.debug(f"Attempting to delete {full_uri} from graphstore")

graphstore_result = graph_delete(
    full_uri, 
    current_app.config["SPARQL_QUERY_ENDPOINT"],
    current_app.config["SPARQL_UPDATE_ENDPOINT"]
)

```

`process_graphstore_record_set` does not just return True or False and will return other things that need to be handled becuase it is designed to handle a batch of updates. We only need the simpler case of a single delete call as we don't have to worry about the other things of rolling back or tracking failures and so on.

Reply:
This is a very good suggestion. Modified the code